File: src/topic_tertule/scripts/person_subscriber.py
#!/usr/bin/env python
# -*- coding:utf-8  -*-
# 该例程将订阅/person_info话题，消息类型topic_tertule::person

import rospy
from topic_tertule.msg import person

def posecallback(msg):
    rospy.loginfo("Subscriber person Info: name:%s age:%d sex:%d",
                    msg.name, msg.age, msg.sex)

def person_subscriber():
    rospy.init_node('person_subscriber', anonymous=True)

    # 创建一个Subscriber，订阅话题/person_info, 注册回调函数posecallback
    rospy.Subscriber("/person_info", person, posecallback)

    # 循环等待回调函数
    rospy.spin()

if __name__ == '__main__':
    person_subscriber()

File: src/Base-src/topic_tertule/src/person_subscriber.cpp
/**
 * 该例程将订阅/person_info的话题,消息类型为topic_tertule::person
 */

 #include<ros/ros.h>
 #include<topic_tertule/person.h>

 // 接收到订阅的消息后，会进入消息回调函数
 void posecallback(const topic_tertule::person::ConstPtr& msg)
 {
     // 接收到的消息打印出来
     ROS_INFO("Subscriber person Info: name:%s age:%d sex:%d",
                 msg->name.c_str(), msg->age, msg->sex);
 }

 int main(int argc, char **argv)
 {
     // 初始化ＲＯＳ节点
     ros::init(argc, argv, "person_subscriber");

     // 创建节点句柄
     ros::NodeHandle n;
     
     // 创建一个subscriber，订阅话题/person_info,注册回调函数posecallback
     ros::Subscriber person_sub = n.subscribe("/person_info", 10, posecallback);

     // 循环等待回调函数
     ros::spin();
          
     return 0;
 }